### DISCUSSION TITLE
Automatically following the route changes ?

### DISCUSSION THREAD

**Discussion by Glideh** [gtm-support/vue-gtm#430] [Q&A] [answered]
Thanks for this nice project
I'm using the version `3.0.1` with Vue `3.3.11`

I didn't get something though:
With my router passed in the `vueRouter` option

```js
app.use(createGtm({
    id: gtmId,
    vueRouter: router
}))
```

Is vue-gtm supposed to track the route changes already ? Or do I need to watch them manually ?

**Comment by Shinigami92** [comment, marked as answer]
See https://github.com/gtm-support/vue-gtm/blob/5268e639f2442f6f0bb2e05f8c7175e73a7826d7/src/index.ts#L121 for details

There is an `afterEach` listener that executes `trackView` on route change

You can also turn on debug option and open browser console

**Comment by Glideh** [comment]
Understood, that's what I was thinking.
So I'm going to try to debug that, thanks.

**Comment by Glideh** [comment]
@Shinigami92 `debug` enabled I'm seeing the _Dispatching Trackview_ with the correct params on the console but I don't see any ajax request going out, is that expected ?
Shouldn't I see a request to google servers on each route change ?

**Reply by Shinigami92** [reply]
1. just enabling debug is not guaranteeing that something works, because `enabling` also is importing

https://github.com/gtm-support/core/blob/1ec51bc62fd192d49f18ad7a65535d19998a1705/src/gtm-support.ts#L170-L171

> If debug mode is enabled, a "Dispatching TrackView" is logged, regardless of whether the plugin is enabled or the plugin is being executed in browser context.

2. I'm not quite sure, I think Google is somehow reading from `window.dataLayer`, but I'm not sure if a call is made 🤔  
   I encourage you to just look a bit more into the source code and explore it, because it is not that much

**Reply by Glideh** [reply]
> because `enabling` also is importing

You mean it should be enabled, right ?
But it's enabled by default isn't it ?

I'll try to check the source too.

**Reply by Glideh** [reply]
Ok I see the conditions, so it should push the event.
https://github.com/gtm-support/core/blob/1ec51bc62fd192d49f18ad7a65535d19998a1705/src/gtm-support.ts#L182-L183
I'll check on the GTM backend, maybe the events are coming.

**Reply by Glideh** [reply]
Yes events are actually incoming, I don't understand how, I don't see any ajax, web socket or other kind of requests. This seems magic :)

**Reply by Shinigami92** [reply]
Google is watching you 😶‍🌫️
